napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,171 @@
+import warnings
+from typing import Union, List, Tuple
+import itertools
+
+import numpy as np
+from napari.layers import Image, Labels
+from vispy.color import Colormap
+from ...utils import colormaps
+
+
+def stack_to_images(
+    stack: Union[Image, Labels],
+    axis: int,
+    colormap: List[Union[str, Colormap]] = None,
+    contrast_limits: List[List[int]] = None,
+    gamma: List[float] = None,
+    blending: str = None,
+    scale: Tuple[float] = None,
+    translate: Tuple[float] = None,
+) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    list
+        List of images
+    """
+
+    data = stack.data
+    name = stack.name
+    num_dim = len(data.shape)
+    n_channels = data.shape[axis]
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting", UserWarning
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
+
+    if colormap is None:
+        if n_channels == 2:
+            colormap = iter(colormaps.MAGENTA_GREEN)
+        if n_channels > 2:
+            colormap = itertools.cycle(colormaps.CYMRGB)
+    else:
+        colormap = iter(colormap)
+
+    if blending not in ['additive', 'translucent', 'opaque']:
+        blending = 'additive'
+
+    if scale is None:
+        scale = np.delete(stack.scale, axis)
+
+    if translate is None:
+        translate = np.delete(stack.translate, axis)
+
+    if contrast_limits is None:
+        contrast_limits = n_channels * [None]
+
+    if gamma is None:
+        gamma = n_channels * [1]
+
+    kwargs = {
+        'rgb': stack.rgb,
+        'blending': blending,
+        'interpolation': stack.interpolation,
+        'rendering': stack.rendering,
+        'iso_threshold': stack.iso_threshold,
+        'attenuation': stack.attenuation,
+        'metadata': stack.metadata,
+        'scale': scale,
+        'translate': translate,
+        'opacity': stack.opacity,
+        'visible': stack.visible,
+        'multiscale': stack.multiscale,
+    }
+
+    imagelist = list()
+
+    for i in range(n_channels):
+        layer_name = f'{name} layer {i}'
+        try:
+            color = next(colormap)
+        except IndexError:
+            color = 'gray'
```

Comment:
I think this try/except can be removed if you implement my suggestions from above.

Reply:
And actually I think this was incorrect, as the above will raise a StopIteration error. Whatever path we decide to adopt, this should be tested!